### ISSUE TITLE
lines-around-comment combined with afterBlockStart throws error

### ISSUE THREAD

**Issue by quintenvk** [eslint/eslint#9342]
The information below is for eslint 4.7.2. I had a slightly different error v3.19.0: `Value "data["0"].afterBlockStart" has additional properties.`

**Tell us about your environment**
Using macOS sierra.

* **ESLint Version: 4.7.2**
* **Node Version: 6.10.0**
* **npm Version: 4.6.1**

**What parser (default, Babel-ESLint, etc.) are you using?**
Using the default parser for ES5 files (unrelated as the issue seems to arise from the eslint config)

**Please show your full configuration:**
Configuration stems from [the docs](https://eslint.org/docs/rules/lines-around-comment).
<details>
<summary>Configuration</summary>

```js
{
    "rules": {
      "lines-around-comment": [
        "error",
        {
          "beforeLineComment": true,
          "afterBlockStart": true
        }
      ]
    }
  }
```

</details>

**What did you do? Please include the actual source code causing the issue.**
Ran `node node_modules/eslint/bin/eslint.js src/js`

**What did you expect to happen?**
The `lines-around-comment`-rule should no longer fail on comments that start without a new line at the beginning of a block.

**What actually happened? Please include the actual, raw output from ESLint.**
The following error occurred
```
/Users/quinten/Sites/domains/mediahuis/frontend-popular/.eslintrc:
	Configuration for rule "lines-around-comment" is invalid:
	Value "[object Object]" should NOT have additional properties.

Error: /Users/quinten/Sites/domains/mediahuis/frontend-popular/.eslintrc:
	Configuration for rule "lines-around-comment" is invalid:
	Value "[object Object]" should NOT have additional properties.

    at validateRuleOptions (/Users/quinten/Sites/domains/mediahuis/frontend-popular/node_modules/eslint/lib/config/config-validator.js:113:15)
    at Object.keys.forEach.id (/Users/quinten/Sites/domains/mediahuis/frontend-popular/node_modules/eslint/lib/config/config-validator.js:153:9)
    at Array.forEach (native)
    at validateRules (/Users/quinten/Sites/domains/mediahuis/frontend-popular/node_modules/eslint/lib/config/config-validator.js:152:30)
    at Object.validate (/Users/quinten/Sites/domains/mediahuis/frontend-popular/node_modules/eslint/lib/config/config-validator.js:230:5)
    at loadFromDisk (/Users/quinten/Sites/domains/mediahuis/frontend-popular/node_modules/eslint/lib/config/config-file.js:549:19)
    at Object.load (/Users/quinten/Sites/domains/mediahuis/frontend-popular/node_modules/eslint/lib/config/config-file.js:592:20)
    at Config.getLocalConfigHierarchy (/Users/quinten/Sites/domains/mediahuis/frontend-popular/node_modules/eslint/lib/config.js:226:44)
    at Config.getConfigHierarchy (/Users/quinten/Sites/domains/mediahuis/frontend-popular/node_modules/eslint/lib/config.js:180:43)
    at Config.getConfigVector (/Users/quinten/Sites/domains/mediahuis/frontend-popular/node_modules/eslint/lib/config.js:285:21)
```

**Comment by not-an-aardvark** [comment]
I think the issue is that your config uses `afterBlockStart`, but the option is actually called `allowBlockStart`.

**Comment by quintenvk** [comment]
That was the error indeed, I can't believe i missed that! Thank you!
